include/circt/Dialect/Emit/EmitOps.td

```diff
@@ -0,0 +1,107 @@
+//===- EmitOps.td - `emit` dialect ops ---------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the MLIR ops for `emit`.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_EMIT_EMITOPS_TD
+#define CIRCT_DIALECT_EMIT_EMITOPS_TD
+
+include "circt/Dialect/HW/HWTypes.td"
+include "circt/Dialect/Emit/EmitDialect.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+class EmitOp<string mnemonic, list<Trait> traits = []> :
+    Op<EmitDialect, mnemonic, traits>;
+
+def FileOp : EmitOp<"file", [
+    Symbol,
+    SingleBlock,
+    NoTerminator,
+    NoRegionArguments,
+    IsolatedFromAbove
+]> {
+  let summary = "Represents the contents of an emitted file";
+
+  let description = [{
+    This operation groups a set of nested operations to be emitted to a file.
+  }];
+
+  let regions = (region SizedRegion<1>:$body);
+  let arguments = (ins
+    StrAttr:$file_name,
+    OptionalAttr<SymbolNameAttr>:$sym_name
```

Comment:
This optionality wasn't in the initial version. Is the idea that we can have a FileOp that doesn't define a symbol, and therefore can't appear in file lists, but otherwise does its usual job? Maybe worth explaining what the symbol is for in the op description.

Reply:
Symbol names are a bit of a pain to conjure up, as they need to be unique, so when we don't need to reference a file, it's best to not have to populate a name.